js/contest.js

```diff
@@ -75,6 +82,44 @@ $(document).ready(function() {
         window.location.href="contest.php?virtual=1";
     });
 
+    function updateTimeSelect(start_time,duration){
+        var lockbefore=Date.parse($("input[name=end_time]").val())-Date.parse($("input[name=lock_board_time]").val());
+        var end_time=new Date();
+        end_time.setTime(start_time+duration*1000);
+        if(end_time.toString()!="Invalid Date") $("input[name=end_time]").val($.format.date(end_time.toString(),"yyyy-MM-dd HH:mm:ss"));
```

Comment:
that won't solve the problem. It's for the case where duration is tooooooooo long.